src/CG/CG_elem.cpp

```diff
@@ -2113,6 +2113,829 @@ bool intersectSegmentPolygon( array3D const &P0, array3D const &P1, std::size_t
+    C = {0.0, 0.0, 0.0};
+    for (std::size_t i = 0; i < nV; ++i) {
+        C += V[i];
+    }
+    C /= double(nV > 0 ? nV : 1);
+
+    //verify the set of vertices is defining a non-zero area polygon
+    array3D N = computePolygonNormal(nV, V);
+    if (norm2(N) <= std::numeric_limits<double>::min()) {
+        throw std::runtime_error ("List of vertices provided do not describe a polygon with area > 0.0. Cannot perform polygon triangulation.");
+        return false;
+    }
+
+    //run the triangulator core and update the sub-triangle structure.
+    subTrianglesCount = 0;
+    if (V) {
+        if (nV < 2 || V == nullptr) {
+            //line and vertex not allowed as polygon or if polygon is not provided properly
+            throw std::runtime_error ("Triangulation of polygon failed.");
+        } else if (nV == 3) {
+            //trivial trangulation, nothing to do.
+            subTrianglesCount = 1;
+            subTrianglesConn  = {{0, 1, 2}};
+        } else {
+            subTrianglesCount = nV;
+            subTrianglesConn.resize(3 * nV);
+
+            for (std::size_t i = 0; i < nV; ++i) {
+                subTrianglesConn[3 * i]     = i;
+                subTrianglesConn[3 * i + 1] = (i + 1) % nV;
+                subTrianglesConn[3 * i + 2] = -1;
+            }
+        }
+    }
+
+    return true;
+}
+
+/*!
+ * Computes polygon intersections with a plane. The polygon is given in a triangulated form.
+ * To triangulate a polygon call method "triangulatePolygon". Intersections are considered only
+ * between the polygonal edges and the plane and not between an edge serving the polygon
+ * triangulation. If intersection occurrs, it can be a single segment (convex, planar polygon)
+ * or a collection of segments. Co-planarity of polygon and plane will be treated as a
+ * no-intersection case. If no intersections occurr, an empty intersection list will be returned.
+ * \param[in] P Point belonging to the plane
+ * \param[in] nP Normal to the plane
+ * \param[in] intersect when false no intersection is occured
+ * \param[in] nV Number  of vertices of the polygon
+ * \param[in] V Pointer to vertices list array. Must have size nV.
+ * \param[in] C Centroid of the polygon used as an auxxiliary point during the triangulation
+ * \param[in] subTrianglesCount number of triangles constituting the polygon
+ * \param[in] subTrianglesConn connectivity of each triangle being part of the triangulated polygon
+ * \param[out] Qs If intersects, return a non-empty list of intersecting Segments.
+ * \param[out] edges_of_Qs If intersects, return the list of polygon edges couple "touched" by the intersecting segment, one for each
+ * intersecting segment.
+ * \param[in] distanceTolerance if distance among features exceed this value they are considered as not intersecting
+ * \return True if intersect, false if not. Return false too if class is not updated.
+ */
+bool intersectPlaneTriPolygon(array3D const &P, array3D const &nP, bool intersect,
```

Comment:
Do you really need this function? It doesn't seem very efficient to triangulate a polygon and then call this function. You can just evaluate the triangulation in place when you need it:
```
    int triangleCount = polygonSubtriangleCount(nV, V);
    for (int triangle=0; triangle < triangleCount; ++triangle) {
        subtriangleOfPolygon( triangle, nV, V, V0, V1, V2);

        ...
    }
```
see for example projectPointPolygon.

Reply:
Rocco didn't use the triangulation of bitpit on purpose, because it couldn't handle all the possible cases like concave polygons if I'm not mistaken.